### PR TITLE
fix possibility of zipping the zip the server is actively zipping

### PR DESCRIPTION
**Pull request by Foxcapades**
Presently, if the formatter is run 2x in the same directory (possibly due to a race condition) the zip code could potentially attempt to zip the file it's presently writing to, recursively zipping the zip file itself infinitely.

### CODE REVIEW

File: blast/formatter/server/go.sum [hunk 1]
```diff
@@ -147,8 +147,6 @@ github.com/tarm/serial v0.0.0-20180830185346-98f6abe2eb07/go.mod h1:kDXzergiv9cb
 github.com/teris-io/shortid v0.0.0-20171029131806-771a37caa5cf/go.mod h1:M8agBzgqHIhgj7wEn9/0hJUZcrvt9VY+Ln+S1I5Mha0=
 github.com/teris-io/shortid v0.0.0-20201117134242-e59966efd125 h1:3SNcvBmEPE1YlB1JpVZouslJpI3GBNoiqW7+wb0Rz7w=
 github.com/teris-io/shortid v0.0.0-20201117134242-e59966efd125/go.mod h1:M8agBzgqHIhgj7wEn9/0hJUZcrvt9VY+Ln+S1I5Mha0=
-github.com/veupathdb/lib-go-blast/v2 v2.0.5 h1:lbyxJyMG6caXnYBAGhxfvufQHOT/mIZj0N03VOLXja0=
```
Comment (Foxcapades) [inline blast/formatter/server/go.sum:150]: auto-generated file.

File: blast/formatter/server/internal/server/run-report-ep.go [hunk 2]
```diff
@@ -82,6 +84,16 @@ func zipDir(path string) (err error) {
...
+	zipPath := filepath.Join(path, reportExportName)
+
+	exists, err := xfiles.FileExists(zipPath)
+	if err != nil {
+		return
+	}
+	if exists {
+		return errors.New(zipPath + " already exists!")
+	}
+
```
Comment (Foxcapades) [inline blast/formatter/server/internal/server/run-report-ep.go:96]: Check for the existence of the zip file before attempting to create a zip file.  If the zip file exists, fail.

File: blast/formatter/server/internal/server/run-report-ep.go [hunk 3]
```diff
@@ -100,11 +112,21 @@ func zipDir(path string) (err error) {
...
+		baseFileName := filepath.Base(matches[i])
+
+		// This should never happen given the check above, but just to be safe!
+		if baseFileName == reportExportName {
+			return errors.New(baseFileName + " already exists!")
+		}
+
+		// Create a file reference in the new zip for the globbed file
+		tf, err = zp.Create(baseFileName)
```
Comment (Foxcapades) [inline blast/formatter/server/internal/server/run-report-ep.go:124]: Additional check (for race condition reasons) to ensure the zip file is not in the globbed file set.

File: blast/formatter/server/internal/xfiles/checks.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package xfiles
+
+import (
+	"errors"
+	"os"
+)
+
+func FileExists(path string) (bool, error) {
```
Comment (Foxcapades) [inline blast/formatter/server/internal/xfiles/checks.go:8]: Simple utility to ensure the file does not already exist.

File: rest-service/Dockerfile [hunk 5]
```diff
@@ -54,7 +54,7 @@ RUN apk add --no-cache tzdata \
 ENV JAVA_HOME=/opt/jdk \
     PATH=/opt/jdk/bin:$PATH \
     XMS=64M \
-    XMX=256M \
+    XMX=1G \
```
Comment (Foxcapades) [inline rest-service/Dockerfile:57]: Bump JVM max memory temporarily until we can resolve the failed queue issue.

File: blast/querier/server/internal/mtx/prometheus.go [hunk 6]
```diff
@@ -25,6 +25,8 @@ var (
 			300,  // 5 Minutes
 			600,  // 10 Minutes
 			900,  // 15 Minutes
+			1200, // 20 Minutes
+			1800, // 30 Minutes
```
Comment (Foxcapades) [inline blast/querier/server/internal/mtx/prometheus.go:29]: Unrelated change caught in the PR.  Adds to more buckets to the blast command time metrics.

File: blast/formatter/server/internal/server/run-report-ep.go [hunk 7]
```diff
@@ -100,11 +112,21 @@ func zipDir(path string) (err error) {
 
 	var tf io.Writer
 	for i := range matches {
-		tf, err = zp.Create(filepath.Base(matches[i]))
+
+		baseFileName := filepath.Base(matches[i])
+
+		// This should never happen given the check above, but just to be safe!
+		if baseFileName == reportExportName {
+			return errors.New(baseFileName + " already exists!")
```
Comment (dmgaldi) [inline blast/formatter/server/internal/server/run-report-ep.go:120]: Where does this error get propagated to? Is there any value in distinguishing the error message in this case as opposed to above case to help debug the race condition?

**Review by dmgaldi — APPROVED** [review]

File: blast/formatter/server/internal/server/run-report-ep.go (hunk 7 above)
Comment (Foxcapades) [reply blast/formatter/server/internal/server/run-report-ep.go:120]: The error is passed up to the caller that throws a 500 error if `err != nil`

File: rest-service/Dockerfile (hunk 5 above)
Comment (rbelnap) [reply rest-service/Dockerfile:57]: I'd suggest leaving this to default to 256M and we can set the var higher in prod temporarily.  (unless a 1G default makes sense as a long-term default)

Comment (Foxcapades) [reply rest-service/Dockerfile:57]: Nope, I'd prefer it to remain at 256M, I'll change it back.
